packages/http/src/instanceWithSpec.ts

```diff
@@ -0,0 +1,41 @@
+import { createInstance } from './index';
+import { getHttpOperationsFromSpec } from './utils/operations';
+import { IHttpConfig, IHttpRequest, ProblemJsonError } from './types';
+import * as pino from 'pino';
+import { pipe } from 'fp-ts/function';
+import { isRight, isLeft } from 'fp-ts/lib/Either';
+import * as TE from 'fp-ts/TaskEither';
+import * as E from 'fp-ts/Either';
+import * as IOE from 'fp-ts/IOEither';
+import { Dictionary } from '@stoplight/types';
+
+export async function createAndCallPrismInstanceWithSpec(
+  document: string | object,
+  options: IHttpConfig,
+  prismRequest: IHttpRequest,
+  logger: pino.Logger
+) {
+  const operations = await getHttpOperationsFromSpec(document);
+  const prism = createInstance(options, { logger: logger.child({ name: 'PRISM INSTANCE' }) });
```

Comment:
can we have the client provide the logger and child we want? that way we dont have to export a pino type. we should have a logger contract in this library, not a 3rd party one

Reply:
its pino all the way down so this does need to remain pino at this time - I have opened an issue here https://github.com/stoplightio/prism/issues/2502 to address this later